docs/streams/developer-guide/processor-api.html

```diff
@@ -108,8 +144,10 @@ <h2><a class="toc-backref" href="#id1">Overview</a><a class="headerlink" href="#
                 times inside <code class="docutils literal"><span class="pre">init()</span></code> method.</p>
             <div class="admonition attention">
                 <p class="first admonition-title"><b>Attention</b></p>
-                <p class="last">Stream-time is only advanced if all input partitions over all input topics have new data (with newer timestamps) available.
-                    If at least one partition does not have any new data available, stream-time will not be advanced and thus <code class="docutils literal"><span class="pre">punctuate()</span></code> will not be triggered if <code class="docutils literal"><span class="pre">PunctuationType.STREAM_TIME</span></code> was specified.
+                <p class="last">Stream-time is only advanced when Streams processes records.
+                  If there are no records to process, or if Streams is waiting for new records
+                  due to the <a class="reference internal" href="/documentation/#streamsconfigs_max.task.idle.ms">Task Idling</a>
+                  configuration, then the stream time will not advance and <code class="docutils literal"><span class="pre">punctuate()</span></code> will not be triggered if <code class="docutils literal"><span class="pre">PunctuationType.STREAM_TIME</span></code> was specified.
```

Comment:
Thanks for pointing this out. It's not right to put a hyphen right here because "time" is a noun and "stream" is an adjective in this usage, as opposed to something like "stream-time punctuation", in which case "punctuation" is the noun, and "stream-time" is effectively a compound word acting as a single adjective.

In some cases, it would be fine to say that "stream-time" could be used here as a compound word functioning as a noun, but for some reason, that doesn't sound right to me in this sentence.

That said, compound word usage in English is pretty idiomatic, and I'm not sure anyone could claim there's a solid rule governing this sentence. All I can say is that "steam time" sounds right to me :)

Reply:
I guess we use `stream-time` as a noun throughout the documentation... But anyway. Not a big deal.